boa_ast/src/temporal/mod.rs

```diff
@@ -0,0 +1,148 @@
+//! AST nodes for Temporal's implementation of ISO8601 grammar.
+
+/// TBD...
+#[derive(Default, Debug)]
+pub struct IsoParseRecord {
```

Comment:
Maybe this should be split into three structs: `IsoDate`, `IsoTime` and `IsoDateTime`. That way, we can return `IsoTime` on `ParseTemporalTimeString` and `IsoDate` on `ParseTemporalMonthDay`/`ParseTemporalYearMonth`.

Reply:
Hmmmm, the one thing that's a bit rough is that `MonthDay` and `YearMonth` can be an `AnnotatedDateTime`, so we would be omitting some of the parsed information if it was lowered to just returning an `IsoDate`.

Reply:
Yes, but [`ParseTemporalMonthDayString`](https://tc39.es/proposal-temporal/#sec-temporal-parsetemporalmonthdaystring) already filters out a lot of info and just returns the date. The same can be said about [`ParseTemporalTimeString`](https://tc39.es/proposal-temporal/#sec-temporal-parsetemporaltimestring) and [`ParseTemporalYearMonth`](https://tc39.es/proposal-temporal/#sec-temporal-parsetemporalyearmonthstring)

Reply:
Hhmm, but we need to throw `RangeError` instead of `ParseError` in the function, so maybe that would complicate things...

Reply:
Ah, we already treat parsing errors as range errors on https://github.com/boa-dev/boa/blob/225eb4aded5a3bb28cbcec34a9e939c9a3dbd3a3/boa_engine/src/builtins/temporal/plain_date/mod.rs#L513. Are all errors thrown on parsing `RangeError`s?

Reply:
Yeah, I believe they are all range errors from what I've seen.

I was more so thinking that the error should be an ISO grammar specific error vs. using the pre-existing `ParseError`. Especially if the parser were potentially to ever move to one from `ICU4X`, then ideally the structure in the built-in may be better aligned. I just haven't gotten to that part yet 😆

I don't see a huge issue in shortening the return into smaller more selective nodes.

Reply:
Cleaned them up 😄 let me know what you think